src/Castle.Core/DynamicProxy/CustomAttributeInfo.cs

```diff
@@ -187,10 +200,15 @@ private static object GetAttributeArgumentValue(Expression arg, bool allowArray)
 						}
 						return array;
 					}
-				}
-				throw new ArgumentException("Only constant and single-dimensional array expressions are supported");
+					break;
 			}
-			return constant.Value;
+			
+			throw new ArgumentException("Only constant and single-dimensional array expressions are supported");
```

Comment:
This message is now a lie. We support expressions that refer to values known only at runtime.

Reply:
Yes, I was just thinking about this. I'll update it.

Reply:
Thanks. I like the new message.